app/controllers/job_invocations_controller.rb

```diff
@@ -52,16 +52,9 @@ def create
 
   def show
     @job_invocation = resource_base.includes(:template_invocations => :run_host_job_task).find(params[:id])
-    @auto_refresh = @job_invocation.task.try(:pending?)
-    @resource_base = @job_invocation.targeting.hosts.authorized(:view_hosts, Host)
-    # There's no need to do the joining if we're not filtering
-    unless params[:search].nil?
-      @resource_base = @resource_base.joins(:template_invocations)
-                                     .where(:template_invocations => { :job_invocation_id => @job_invocation.id})
-    end
-    @hosts = resource_base_search_and_page
     @job_organization = Taxonomy.find_by(id: @job_invocation.task.input[:current_organization_id])
     @job_location = Taxonomy.find_by(id: @job_invocation.task.input[:current_location_id])
+    targeting_hosts_resources if params[:format] == 'json'
```

Comment:
just a nit - could we just use `respond_to do |format|` [pattern](https://api.rubyonrails.org/classes/ActionController/MimeResponds.html#method-i-respond_to)?